src/main/java/com/google/sps/data/AvailabilityTimeSlotGenerator.java

```diff
@@ -0,0 +1,160 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.TextStyle;
+import java.util.Locale;
+import java.time.format.DateTimeFormatter;
+
+// TODO: Add a test for the constructor.
+
+/** A generator of a collection of AvailabilityTimeSlot Objects. */
+public class AvailabilityTimeSlotGenerator {
+  private static final int numberOfSlotsPerDay = 48;
+  private static final int TIME_8AM = 8;
+  private static final int TIME_7PM = 19;
+
+  /**
+   * Constructs list of AvailabilityTimeSlot objects
+   *
+   * @param timezoneOffset A String that represents the difference between UTC and the user's
+   *     current timezone. Example: A user in EST has a timezoneOffset of "-240" which means that
+   *     EST is 240 minutes behind UTC.
+   */
+  public static List<AvailabilityTimeSlot> getTimeSlots(String timezoneOffset) {
+    ZoneOffset timeZoneOffset = convertStringToOffset(timezoneOffset);
+    ZoneId zoneId = ZoneId.ofOffset("UTC", timeZoneOffset);
+    ZonedDateTime today = Instant.now().atZone(zoneId);
+    // TODO: Create a timeSlotsForWeek(today) method.
+    return timeSlotsForDay(today);
+  }
+
+  /**
+   * Generates a day's worth of time slots.
+   *
+   * @param today The day that time slots are generate for.
```

Comment:
There seems to be an implicit requirement that `today` is in the UTC zone, which needs to be documented. It might be better to accept an `Instant` that is in the current day and generate slots for that. This would mean that the code of `getTimeSlots` above would just be `timeSlotsForDay(Instant.now(), timezoneOffset)`, meaning that it could probably be eliminated (removing the need to test it).

Reply:
Done